### ISSUE TITLE
Support \Q\E quotation

### ISSUE THREAD

**Issue by vkopichenko** [mifmif/Generex#22]
It would be nice to provide possiblity to escape some parts of regex from generation without need to escape each special character one by one.
For example I'd like to use
> minion_\d{3}\Q@gru.evil\E

instead of 

> minion_\d{3}\@gru\\.evil

Here is a workaround code to transform the former to latter:
```
    private static String requote(String regex) {
        final Pattern PATTERN_QUOTED = Pattern.compile("\\\\Q(.*?)\\\\E");
        // http://stackoverflow.com/questions/399078/what-special-characters-must-be-escaped-in-regular-expressions
        // adding "@" prevents StackOverflowError inside generex: https://github.com/mifmif/Generex/issues/21
        final Pattern PATTERN_SPECIALS = Pattern.compile("[.^$*+?(){|\\[\\\\@]");
        StringBuilder sb = new StringBuilder(regex);
        Matcher matcher = PATTERN_QUOTED.matcher(sb);
        while (matcher.find()) {
            sb.replace(matcher.start(), matcher.end(), PATTERN_SPECIALS.matcher(matcher.group(1)).replaceAll("\\\\$0"));
            //matcher.reset();
        }
        return sb.toString();
    }
```

**Comment by mifmif** [comment]
It's a nice feature, I will work on this point, you can fork the project and add this feature, then i will merge your request.

**Comment by mifmif** [comment]
This feature is now supported

**Comment by vkopichenko** [comment]
The initial requote implementation here were vulnerable to some problems:
* no escape for &"<#~ automaton special chars
* incorrect handling of multiple adjacent quote groups
* no support for multiline patterns

These were fixed in PR #48.
